### PR TITLE
Refs #6923 Test refactor

### PR DESCRIPTION
**Pull request by LuisGP**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -35,7 +35,17 @@ target_compile_features(${PROJECT_NAME}
 #####################################################################################
 #                                    Examples
 #####################################################################################
-macro(compile_example EXAMPLE_NAME EXAMPLE_SOURCE)
+macro(compile_example)
```
Comment (lemunozm) [inline CMakeLists.txt:38]: Cool!

File: include/xtypes/DynamicData.hpp [hunk 2]
```diff
@@ -652,10 +652,17 @@ class WritableDynamicDataRef : public ReadableDynamicDataRef
...
     /// \pre The DynamicData must represent a SequenceType.
+    /// \pre The bounds must be greater or equal to the new size.
     /// \returns The writable reference to this DynamicData
     WritableDynamicDataRef& resize(size_t size) // this = SequenceType
     {
         assert(type_.kind() == TypeKind::SEQUENCE_TYPE);
+        assert(bounds() >= size); // Cannot grow over the bounds
+        if (bounds() < size)
+        {
+            // Release protection.
```
Comment (lemunozm) [inline include/xtypes/DynamicData.hpp]: Define "protection". How does this protect the user despite its cost?

File: include/xtypes/SequenceInstance.hpp [hunk 3]
```diff
@@ -49,7 +49,7 @@ class SequenceInstance
         : content_(other.content_)
         , block_size_(other.block_size_)
         , capacity_(other.capacity_)
-        , memory_(other.capacity_ > 0 ? new uint8_t[other.capacity_ * other.block_size_] : nullptr)
+        , memory_(capacity_ > 0 ? new uint8_t[capacity_ * block_size_] : nullptr)
```
Comment (lemunozm) [inline include/xtypes/SequenceInstance.hpp:51]: :+1:

File: test/unitary/xtypes/consistency.cpp [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+
+    EXPECT_EQ(TypeConsistency::IGNORE_SEQUENCE_BOUNDS |
+              TypeConsistency::IGNORE_TYPE_SIGN, a_seq.is_compatible(b_seq));
+
+    EXPECT_EQ(TypeConsistency::IGNORE_ARRAY_BOUNDS |
+              TypeConsistency::IGNORE_TYPE_WIDTH |
+              TypeConsistency::IGNORE_TYPE_SIGN , a_arr.is_compatible(b_arr));
+
+    a.add_member(
+            Member("a_string", a_string)).add_member(
```
Comment (lemunozm) [inline test/unitary/xtypes/consistency.cpp]: I let you to adapt it to the obvious format (see similar occurrences)

File: test/unitary/xtypes/struct_type.cpp [hunk 5]
```diff
@@ -0,0 +1,419 @@
...
+
+static const float STRUCTS_SIZE = 1E1;
+
+/**********************************
+ *        StructType Tests        *
+ **********************************/
+
+TEST (StructType, primitive_struct)
+{
+    StructType st("struct_name");
```
Comment (lemunozm) [inline test/unitary/xtypes/struct_type.cpp:50]: This is pending to refactorize from Francesco, right?

**Review by lemunozm — CHANGES_REQUESTED** [review]

File: include/xtypes/DynamicData.hpp (hunk 2 above)
Comment (LuisGP) [reply include/xtypes/DynamicData.hpp]: It protects the user by doing nothing. Just returns the sequence without changes.

File: test/unitary/xtypes/struct_type.cpp (hunk 5 above)
Comment (LuisGP) [reply test/unitary/xtypes/struct_type.cpp:50]: Added some helper functions to ease code reading.

File: include/xtypes/DynamicData.hpp (hunk 2 above)
Comment (lemunozm) [reply include/xtypes/DynamicData.hpp]: And, is this really a protection? I mean, if the API is used bad, why we need to protect them? With a cost in efficiency?

File: test/unitary/xtypes/struct_type.cpp (hunk 5 above)
Comment (lemunozm) [reply test/unitary/xtypes/struct_type.cpp:50]: Great!

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/18/)

File: include/xtypes/DynamicData.hpp (hunk 2 above)
Comment (LuisGP) [reply include/xtypes/DynamicData.hpp]: Removed the protection, this checking will be performed by extended behavior of `xtypes_assert`.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/20/)

File: include/xtypes/DynamicData.hpp (hunk 2 above)
Comment (lemunozm) [reply include/xtypes/DynamicData.hpp]: Cool :+1:

**Review by lemunozm — APPROVED** [review]

**Pull request merged**